scanner/indexer/indexer.go

```diff
@@ -76,54 +77,79 @@ type Indexer interface {
 // localIndexer is the Indexer implementation that runs libindex locally.
 type localIndexer struct {
 	libIndex        *libindex.Libindex
+	pool            *pgxpool.Pool
+	root            string
 	getLayerTimeout time.Duration
 }
 
 // NewIndexer creates a new indexer.
 func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "scanner/backend/indexer.NewIndexer")
+
 	pool, err := postgres.Connect(ctx, cfg.Database.ConnString, "libindex")
 	if err != nil {
 		return nil, fmt.Errorf("connecting to postgres for indexer: %w", err)
 	}
+	defer func() {
+		if err != nil {
+			pool.Close()
+		}
+	}()
+
 	store, err := postgres.InitPostgresIndexerStore(ctx, pool, true)
 	if err != nil {
 		return nil, fmt.Errorf("initializing postgres indexer store: %w", err)
 	}
+	defer utils.IgnoreError(func() error {
+		if err != nil {
+			return store.Close(ctx)
+		}
+		return nil
+	})
+
```

Comment:
I've come to appreciate the direct style as in:

```suggestion
	defer func() {
		if err != nil {
			_ = store.Close(ctx)
		}
	}()
```

I agree we are mixing it up now, so fine to leave it. But given this is a clean-up PR, would you like to move out from `utils.IgnoreError()`?

Reply:
👍